src/components/toolbox/autoSuggest/index.js

```diff
@@ -117,62 +115,57 @@ class AutoSuggest extends React.Component {
     this.loaderTimeout = setTimeout(() => {
       // istanbul ignore else
       if (this.getFilterList().length >= 0) this.setState({ isLoading: false });
-      this.props.validateBookmark();
+      this.props.onChangeDelayed();
     }, 300);
 
     if (e && e.target && e.target.value === '') this.resetListIndex();
     this.props.onChange(e);
   }
 
-  // eslint-disable-next-line complexity
   render() {
     const {
-      recipient,
+      selectedItem,
       placeholder,
+      renderItem,
+      renderIcon,
+      className,
     } = this.props;
     const { dropdownIndex } = this.state;
-    const haveAvatar = recipient.address.length && !recipient.error;
-    const selectedAccount = recipient.selected ? recipient.title : recipient.value;
 
     return (
       <Fragment>
-        <span className={`${styles.recipientField} recipient`}>
-          <AccountVisual
-            className={styles.accountVisual}
-            address={recipient.address}
-            placeholder={!haveAvatar}
-            size={25}
-          />
+        <span className={`${styles.inputWrapper} ${className}`}>
+          <span className={styles.icon}>
+            {renderIcon(selectedItem)}
+          </span>
           <Input
             autoComplete="off"
-            className={`${styles.input} ${recipient.error ? 'error' : ''} recipient bookmark`}
-            name="recipient"
-            value={selectedAccount}
+            className={`${styles.input} ${selectedItem.error ? 'error' : ''} ${className} bookmark`}
+            name={className}
+            value={selectedItem.selected ? selectedItem.title : selectedItem.value}
             placeholder={placeholder}
             onKeyDown={this.onHandleKeyPress}
             onChange={this.onChange}
           />
-          <Spinner className={`${styles.spinner} ${this.state.isLoading && recipient.value ? styles.show : styles.hide}`} />
+          <Spinner className={`${styles.spinner} ${this.state.isLoading && selectedItem.value ? styles.show : styles.hide}`} />
           <Icon
-            className={`${styles.status} ${!this.state.isLoading && recipient.value ? styles.show : styles.hide}`}
-            name={recipient.error ? 'alertIcon' : 'okIcon'}
+            className={`${styles.status} ${!this.state.isLoading && selectedItem.value ? styles.show : styles.hide}`}
+            name={selectedItem.error ? 'alertIcon' : 'okIcon'}
           />
-          <div className={`${styles.bookmarkContainer}`}>
+          <div className={`${styles.listContainer}`}>
             <div ref={(node) => { this.listContainerRef = node; }}>
-              <ul className={`${styles.bookmarkList} bookmark-list`}>
+              <ul className={`${styles.itemList} bookmark-list`}>
```

Comment:
Maybe would be worth to move this to an `AutoSuggest.Suggestions` component, to use similar as the other dropdown components? And would also simplify the props that needs to be passed to the `AutoSuggest` component